analytics/spark/src/main/java/org/locationtech/geowave/analytic/spark/RDDUtils.java

```diff
@@ -189,43 +199,86 @@ public static InsertionIds trimIndexIds(
+    final Broadcast<String> typeName = sc.broadcast(adapter.getTypeName(), stringTag);
+    final Broadcast<String> indexName = sc.broadcast(index.getName(), stringTag);
 
     // map to a pair containing the output key and the output value
     inputRDD.mapToPair(
-        feat -> new Tuple2<GeoWaveOutputKey, SimpleFeature>(
+        feat -> new Tuple2<>(
             new GeoWaveOutputKey(typeName.value(), indexName.value()),
             feat)).saveAsNewAPIHadoopDataset(job.getConfiguration());
   }
 
+  private static void writeRasterToGeoWave(
+      final SparkContext sc,
+      final Index index,
+      final DataStorePluginOptions outputStoreOptions,
+      final DataTypeAdapter adapter,
+      final JavaRDD<GridCoverage> inputRDD) throws IOException {
+
+    // setup the configuration and the output format
+    final Configuration conf = new org.apache.hadoop.conf.Configuration(sc.hadoopConfiguration());
+
+    GeoWaveOutputFormat.setStoreOptions(conf, outputStoreOptions);
+    GeoWaveOutputFormat.addIndex(conf, index);
+    GeoWaveOutputFormat.addDataAdapter(conf, adapter);
+
+    // create the job
+    final Job job = new Job(conf);
+    job.setOutputKeyClass(GeoWaveOutputKey.class);
+    job.setOutputValueClass(GridCoverage.class);
+    job.setOutputFormatClass(GeoWaveOutputFormat.class);
+
+    // broadcast string names
+    final ClassTag<String> stringTag = scala.reflect.ClassTag$.MODULE$.apply(String.class);
+    final Broadcast<String> typeName = sc.broadcast(adapter.getTypeName(), stringTag);
+    final Broadcast<String> indexName = sc.broadcast(index.getName(), stringTag);
+
+    // map to a pair containing the output key and the output value
+    inputRDD.mapToPair(
+        gridCoverage -> new Tuple2<>(
+            new GeoWaveOutputKey(typeName.value(), indexName.value()),
+            gridCoverage)).saveAsNewAPIHadoopDataset(job.getConfiguration());
+  }
+
   public static Broadcast<? extends NumericIndexStrategy> broadcastIndexStrategy(
-      SparkContext sc,
-      NumericIndexStrategy indexStrategy) {
-    ClassTag<NumericIndexStrategy> indexClassTag =
+      final SparkContext sc,
+      final NumericIndexStrategy indexStrategy) {
+    final ClassTag<NumericIndexStrategy> indexClassTag =
         scala.reflect.ClassTag$.MODULE$.apply(indexStrategy.getClass());
-    Broadcast<NumericIndexStrategy> broadcastStrategy = sc.broadcast(indexStrategy, indexClassTag);
+    final Broadcast<NumericIndexStrategy> broadcastStrategy =
+        sc.broadcast(indexStrategy, indexClassTag);
     return broadcastStrategy;
   }
+
+  public static void writeRasterRDDToGeoWave(
+      final SparkContext sc,
+      final Index index,
+      final DataStorePluginOptions outputStoreOptions,
+      final RasterDataAdapter adapter,
+      final JavaRDD<GridCoverage> inputRDD) throws IOException {
+
+    writeRasterToGeoWave(sc, index, outputStoreOptions, adapter, inputRDD);
```

Comment:
Is there a need for the private method to exist?  It has the same signature as the public one.

Reply:
good catch, I think this went through several revisions and with what was left the extraneous method makes no sense